### ISSUE TITLE
Duplicated cursors and fullscreen mismatching (with bonus display settings bug)

### ISSUE THREAD

**Issue by ltsymes23** [pop-os/cosmic-comp#871]
After commenting on a different bug regarding fullscreen game mouse usage [https://github.com/pop-os/cosmic-comp/issues/484#issuecomment-2361042652], I found something much more interesting.

Original comment:
> The game I noticed this take place in Genshin Impact running the hoyoplay launcher as a non-steam game under proton. Pre nvidia 560 driver release, clicks would not register at all meaning you would not be able to start the game, but now with the 560 driver update clicking now works, but the click takes place 10 cms to the right of where the cursor icon appears. This means the left side of the screen is inaccessible.

I realised after some more experimentation that there is a representation of that second invisible cursor behind the game which could be seen on a second monitor, and not only that, when messing with the resolutions I managed to do this on the cosmic desktop itself. Behold...

![20240919_232936](https://github.com/user-attachments/assets/cd59319c-884b-4b4d-8f2c-835b264d66d7)

I have replicated this multiple times by changing the resolution between native 1440p and 1080p (which had some unusual interactions between the resolution and the scaling settings I will make another bug report for) resulting in the cursor being visible across 2 monitors, where the cursor on the right takes precedent over the left, the same as how the left side of the monitor was inaccessible ingame.

GPU: NVIDIA GeForce GTX 1080 Ti
Driver Version: 560.35.03

cosmic-comp:
  Installed: 0.1~1726667039~24.04~7c24e36
  Candidate: 0.1~1726667039~24.04~7c24e36
  Version table:
 *** 0.1~1726667039~24.04~7c24e36 1002
       1001 http://apt.pop-os.org/release noble/main amd64 Packages
       1002 http://apt.pop-os.org/staging/master noble/main amd64 Packages
        100 /var/lib/dpkg/status

**Comment by mmstick** [comment]
Make sure to align display positions to correct any overlap.

**Comment by ltsymes23** [comment]
That was going to be my next report which I just found out might be the root of the whole issue, screenshots coming

**Comment by ltsymes23** [comment]
Changes to the display settings are not reflected until exiting settings or switching tabs. 1 is after settings have been changed on the middle monitor from 125% scale to 100% scale, and 2 is after closing and opening the settings app which shows that the displays are overlapping and reproducing the above issue of multiple cursors

1.
![image](https://github.com/user-attachments/assets/8a87cca2-9c1e-4cd5-88eb-c5aa3fff9861)

2.
![image](https://github.com/user-attachments/assets/3aa81834-4656-4fee-8d27-78207b7bf7f4)


If the game automatically scales the resolution of the game, overlaps occurring would explain it

**Comment by mmstick** [comment]
If a game changes resolution, cosmic-comp is where you should look for this issue

**Comment by ltsymes23** [comment]
I tried with my monitors at 1080p and now the cursor is on the opposite side to before, so now I can only click on the right side of the screen. I could access the settings menu now though so I could check that the game resolution was at 1080p as well so its not as simple as changing resolution unfortuantely

**Comment by feikedonia** [comment]
I have a possible duplicate issue: #1075

**Comment by zsilet** [comment]
The issue @feikedonia mentioned just happened to me (video linked below). The really unfortunate part about the issue is that it is somewhat unrecoverable (for mere mortals), as in my case, mouse clicks were registering on the display opposite of the display settings. 

@git-f0x kindly pointed out that I can delete the outputs.ron file, which, along with an immediate forced poweroff, helped me recover. 

I tried to reproduce it for a while, but I cannot figure out how I ended up with overlapping displays. At the time the issue happened, I was fighting with a different (recently introduced issue), whereby after turning off the monitor (due to inactivity, before the computer sleeps), the external HDMI display will not come back on. I was in the process of toggling the external monitor to disabled/enabled, when the two ended up on top of each other. 

https://photos.app.goo.gl/U5Q55uWGHBBrqjpw7

**Comment by Drakulix** [comment]
> If a game changes resolution, cosmic-comp is where you should look for this issue

A game can't force a specific resolution on wayland. That just affects it's inner render resolution, so this is a complicated mess of potentially a cosmic-settings/cosmic-randr issue causing the overlap and cosmic-comp allowing the overlap. (I should code something up to sanitize the values to the point where overlaps of any kind are impossible....)

**Comment by SLASHLogin** [comment]
I'm experiencing the same issue and can easily replicate it with my setup. It only requires me to disable display 0 in the settings, and then enable it again, which causes the displays to overlap with each other.

When this happens, I can see one cursor, which registers clicks on the other display, similarly to others' issues.

My setup:

- GPU: NVIDIA GeForce RTX 3070 Ti
- Driver version: 570.133.07
- cosmic-comp version: 0.1\~1745930527\~24.04\~d7b37cf

I'll be happy to provide more information or test potential fixes if needed.

![Image](https://github.com/user-attachments/assets/6170d0f8-5deb-4059-93be-db6adaceb6d6)

![Image](https://github.com/user-attachments/assets/03fdf104-7ac8-4888-8c8e-e14c3bc6d972)

**Comment by SLASHLogin** [comment]
I've noticed that there was a very similar issue created https://github.com/pop-os/cosmic-comp/issues/1592. It got resolved in the following PR https://github.com/pop-os/cosmic-comp/pull/1598.

Ever since using versions that include this change, I'm unable to replicate the issue. However, there seem to be moments where cosmic-comp crashes completely, and I'm left with a black screen, but this might be unrelated.

Additionally, there was this comment attached to the PR https://github.com/pop-os/cosmic-comp/pull/1598#pullrequestreview-3137648216, which suggests that it's still possible to end up in the invalid state.

Thus, I believe this issue could be closed if everyone else can't reproduce the issue on their side.

**Comment by LiamillionSS** [comment]
I agree, original issue I posted about cannot be replicated. Will open issue again if needed but remaining problems would likely be much more specific. Thanks all
